### PR TITLE
Release 2.0

### PR DESCRIPTION
**Pull request by vlad-zhukov**
# ~~WIP, DON'T MERGE~~

Here we are working on a new major release according to the [2.0 roadmap](https://github.com/andywer/webpack-blocks/issues/264). Use this branch as the base for new PRs.

### CODE REVIEW

File: docs/BLOCK-CREATION.md [hunk 1]
```diff
@@ -108,13 +108,7 @@ function sampleBlock () {
...
 The context object is a metadata object that is passed to every block. It is meant to contain any kind of data that is needed for webpack config creation, but not part of the webpack config itself.
 
-Initially it will contain the `fileType` mapping and a `webpack` instance. If you are using [hooks](#hooks) you might want to put custom metadata into the context and use it in the `post` hook.
-
-### context.fileType
-
-*Deprecated. This feature will be removed soon. Use `match()` and `context.match` instead.*
-
-The `context.fileType` is a mapping from MIME type (`application/javascript`, `text/css`, ...) to a regular expression used for matching filenames. You can find the default file types and the extensions they match [here](https://github.com/andywer/webpack-blocks/blob/master/packages/core/lib/defaultFileTypes.js).
+Initially it will contain a `webpack` instance. If you are using [hooks](#hooks) you might want to put custom metadata into the context and use it in the `post` hook.
```
Comment (andywer) [inline docs/BLOCK-CREATION.md]: 👌

**Comment by andywer** [comment]
<p align=center>
  <b>I can't tell you how much I like this PR 😊</b>
</p>

bildschirmfoto 2018-04-21 um 16 31 04

**Comment by andywer** [comment]
PS: Changed the PR title.

**Comment by andywer** [comment]
Did someone change the PR title again? "Release 2.0" is not a good title for this PR...

**Comment by vlad-zhukov** [comment]
This is a tracking PR of all v2 changes 😁 Don't bother reviewing it for now, it will be ready when all 2.0.0 milestone are done!

**Comment by andywer** [comment]
@vlad-zhukov Why not merge this already as "Remove deprecated features"? Multiple short PRs are easier to review and comprehend than few large ones :)

**Comment by vlad-zhukov** [comment]
Because these changes have been already reviewed and merged into this branch. I am not going to push commits into this branch directly, only through PRs! 😏

**Comment by andywer** [comment]
Ahhh, now I get it 😉

I would suggest, though, that we either
a) close this PR, just keep merging into this release branch and later open the PR when we are ready to merge
or
b) Add in BIG letters to the title: `Release 2.0. WIP. DON'T MERGE YET.`

So no one can accidentally merge this prematurely. You know, accidents happen... 😅
We could also create a `Version 2.0` milestone and add all PRs / issues to it, so we can easily track them.

Thoughts?

**Comment by vlad-zhukov** [comment]
I would like to have this PR opened to keep track of certain stuff. Added a WIP note, is it ok?

And I've already created a 2.0.0 milestone earlier today.

**Comment by andywer** [comment]
Ok. Just did a "request changes" to prevent merging.

Btw, thank you guys for your efforts! You are really moving the project forward 😊

**Comment by Lavoaster** [comment]
Is there any chance of adding a version property to the `package.json` file? I'm trying to import this branch on a project to update to webpack 4.0 (and hopefully help test some of these changes) but yarn is throwing a fit at the lack of a version. 

```
"webpack-blocks": "https://github.com/andywer/webpack-blocks.git#release-2.0",
```

**Comment by andywer** [comment]
@Lavoaster Yeah, that doesn‘t work with a monorepo... But I agree. We should trim down the v2.0 roadmap and release ASAP for webpack 4‘s sake.

@vlad-zhukov Is there anything stopping us from releasing this branch as is right now as v2.0.0-rc?

**Comment by vlad-zhukov** [comment]
There are a few planned breaking changes left in #255:

- webpack-dev-server has been deprecated in favour of webpack-serve
- extract-text-webpack-plugin is deprecated for CSS in favour of mini-css-extract-plugin
- (also perhaps babel 7 if it goes out of beta?)

Are we fine with releasing a v2.0.0-alpha.1 for testing and merge a few breaking changes later? If so I'd like to update deps first, will send a PR today.

**Comment by andywer** [comment]
Sounds good 👍

**Comment by vlad-zhukov** [comment]
Ready for the 1st alpha I guess!
@andywer would you like to merge this or release a new version from this branch?

**Comment by andywer** [comment]
Sure! Can do, but not before tonight 😉

**Comment by andywer** [comment]
Sorry, didn't realize it was a question... 😅

Let's publish from this feature branch for the alpha release.

**Comment by andywer** [comment]
Published as `v2.0.0-alpha`, npm tag `next`. Install as `npm i --save-dev webpack-blocks@next`.

@Lavoaster

**Comment by Lavoaster** [comment]
Thanks @andywer @vlad-zhukov 

Upgraded this and to webpack 4.0 the other day and it all went pretty smooth, all I really needed to do was add the `mode` block to my config. So looks good here right now.

**Comment by vpsouza** [comment]
@Lavoaster can you help me with a sample config for webpack 4.0?

**Comment by dmitmel** [comment]
@vpsouza Do you mean a sample config with webpack-blocks? If so, I have [one](https://github.com/dmitmel/quiz9000/blob/master/config/webpack.config.js) in my project. It's a refactored version of the [webpack config from create-react-app](https://github.com/facebook/create-react-app/tree/next/packages/react-scripts/config). Although it's not perfect and it uses webpack-blocks v1.0, you may find it useful.

**Comment by andywer** [comment]
@dmitmel Sure. Feel free to share it in the gitter chat room :)

**Pull request merged**